### ISSUE TITLE
padded blocks should allow single lines.

### ISSUE THREAD

**Issue by allanchau** [eslint/eslint#11449]
I've been following https://github.com/eslint/eslint/issues/7145 and other similar issues and thought the option `allowSingleLineBlocks` would fix this. I am actually looking for a rule which:
- Disallows padding (new lines) if the code within the block is only a single line.
- Force padding if the code within the block spans multiple lines.

**What rule do you want to change?**

padded-blocks

**Does this change cause the rule to produce more or fewer warnings?**

Fewer.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option.

**Please provide some example code that this change will affect:**

```js
// correct
if (cond) {
    doSomething();
}

if (cond) {

    doSomething()
        .then(...)
        .catch(...);

}

// incorrect
if (cond) {

    doSomething();

}

if (cond) {
    doSomething()
        .then(...)
        .catch(...);
}
```

**What does the rule currently do for this code?**

Warn or nothing depending on the settings.

**What will the rule do after it's changed?**

Produce less warnings.

**Are you willing to submit a pull request to implement this change?**

Sure, but I wouldn't know where to get started 😄 .
